test/Dapr.Actors.Test/ApiTokenTests.cs

```diff
@@ -0,0 +1,175 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Dapr.Actors.Client;
+using FluentAssertions;
+using Xunit;
+
+namespace Dapr.Actors.Test
+{
+    public class ApiTokenTests : IDisposable
+    {
+        public void Dispose()
+        {
+            Environment.SetEnvironmentVariable("DAPR_API_TOKEN", "");
+        }
+        
+        [Fact]
+        public void CreateProxyWithRemoting_WithApiToken()
+        {
+            var actorId = new ActorId("abc");
+            var handler = new TestHttpClientHandler();
+            var options = new ActorProxyOptions
+            {
+                DaprApiToken = "test_token",
+                HttpClientMessageHandler = handler
+            };
+            var factory = new ActorProxyFactory(options);
+            var proxy = factory.CreateActorProxy<ITestActor>(actorId, "TestActor");
+            var task = proxy.SetCountAsync(1, new CancellationToken());
+
+            handler.Requests.TryDequeue(out var entry).Should().BeTrue();
+            var headerValues = entry.Request.Headers.GetValues("dapr-api-token");
+            headerValues.Should().Contain("test_token");
+        }
+
+        [Fact]
+        public void CreateProxyWithRemoting_WithApiTokenEnvVarSet()
+        {
+            var actorId = new ActorId("abc");
+            var handler = new TestHttpClientHandler();
+            var options = new ActorProxyOptions
+            {
+                HttpClientMessageHandler = handler
+            };
+            Environment.SetEnvironmentVariable("DAPR_API_TOKEN", "test_token");
```

Comment:
This is the reason why I want to see all of the defaulting logic pulled out of all of the implementation code. if you have one-two places that read the environment variable, and all of the complex parts of the system just trust the data they are given then it's really easy to understand and test.